picdar-export/src/main/scala/com.gu.mediaservice.picdarexport/lib/db/ExportDynamoDB.scala

```diff
@@ -154,17 +183,61 @@ class ExportDynamoDB(credentials: AWSCredentials, region: Region, tableName: Str
       dateRange.end.map(asRangeString).map(":endDate" -> _)
   }
 
+  def getRow(ref: AssetRef) = Future {
+    table.getItem(
+      new GetItemSpec().
+      withPrimaryKey(
+        IdKey, ref.urn,
+        RangeKey, PicdarDates.format.print(ref.dateLoaded)
+      )
+    )
+  }
+
+  def getUrnsForDateRange(dateRange: DateRange) = Future {
+    val imageIndex = table.getIndex(picdarCreatedIndex)
+
+    val items = for {
+      date  <- dateRange.dateList
+
+      dateString = PicdarDates.format.print(date)
+      key = new KeyAttribute(RangeKey, dateString)
+      query = imageIndex.query(key).pages.asScala
+
+      pages <- query
+      items <- pages
+    } yield items
+
+    items.map(AssetRef(_))
+  }
+
+  // Type this method so we can coerce items into whatever you want!
+  import com.amazonaws.services.dynamodbv2.document.Item
+  def getUrnsForNotFilledFields[T](dateRange: DateRange, attrs: Set[String])(f: Item => T): Future[Seq[T]] = {
```

Comment:
I almost did it here, but I feel like it's probably too big a change for this PR. I'd like to try this out in PROD a bit then make the change when we're confident this approach performs well.